opentracing-shim/src/opentracingshim/__init__.py

```diff
@@ -0,0 +1,248 @@
+from opentracingshim import util
+
+
+def create_tracer(tracer: Tracer) -> opentracing.Tracer:
+    return TracerWrapper(tracer)
+
+
+class SpanContextWrapper(opentracing.SpanContext):
+    def __init__(self, otel_context: SpanContext):
+        self._otel_context = otel_context
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `SpanContext` object."""
+
+        return self._otel_context
+
+    @property
+    def baggage(self):
+        return {}
+        # TODO: Implement.
+
+
+class SpanWrapper(opentracing.Span):
+    def __init__(self, tracer, context, span: Span):
+        self._otel_span = span
+        opentracing.Span.__init__(self, tracer, context)
+
+    def unwrap(self):
+        """Returns the wrapped OpenTelemetry `Span` object."""
+
+        return self._otel_span
+
+    @property
+    def context(self):
+        return self._context
+
+    @property
+    def tracer(self):
+        return self._tracer
+
+    def set_operation_name(self, operation_name):
+        self._otel_span.update_name(operation_name)
+        return self
+
+    def finish(self, finish_time=None):
+        self._otel_span.end()
+        # TODO: Handle finish_time. The OpenTelemetry API doesn't currently
+        # support setting end time on a span and we cannot assume that all
+        # OpenTelemetry Tracer implementations have an `end_time` field.
+        # https://github.com/open-telemetry/opentelemetry-python/issues/134
+
+    def set_tag(self, key, value):
+        self._otel_span.set_attribute(key, value)
+
+    def log_kv(self, key_values, timestamp=None):
+        if timestamp is None:
+            timestamp = util.time_ns()
+
+        event_name = util.event_name_from_kv(key_values)
+        event = Event(event_name, timestamp, key_values)
+        self._otel_span.add_lazy_event(event)
+
+        # Return self for call chaining.
+        return self
+
+    def set_baggage_item(self, key, value):
+        return self
+        # TODO: Implement.
+
+    def get_baggage_item(self, key):
+        return None
+        # TODO: Implement.
+
+    # TODO: Verify calls to deprecated methods `log_event()` and `log()` on
+    # base class work properly (it's probably fine because both methods call
+    # `log_kv()`).
+
+
+class ScopeWrapper(opentracing.Scope):
+    def __init__(self, manager, span):
+        # pylint: disable=super-init-not-called
+        self._manager = manager
+        self._span = span
+
+    @property
+    def span(self):
+        return self._span
+
+    @property
+    def manager(self):
+        return self._manager
+
+    def close(self):
+        self._span.finish()
+        # TODO: Set active span on OpenTelemetry tracer.
+        # https://github.com/open-telemetry/opentelemetry-python/issues/161#issuecomment-534136274
+
+
+class ScopeManagerWrapper(opentracing.ScopeManager):
+    def __init__(self, tracer: "TracerWrapper"):
+        # pylint: disable=super-init-not-called
```

Comment:
I'm largely a python ignoramus, but I wonder if this thing shouldn't have a comment about why this linter warning should be ignored.

This is same for other cases of linter warning ignore clauses.

Reply:
Excellent point. I've gone through similar cases and it turns out we *did* have to call `super().__init__()` on some cases. On other cases we don't even need to override the `__init__()` method from the base class. I'll push a cleanup in a bit.